subprojects/core/src/main/java/org/gradle/api/internal/AsmBackedClassGenerator.java

```diff
@@ -407,7 +407,7 @@ public void addPropertySetters(PropertyMetaData property, Method getter) throws
             //    ((PropertyInternal)<getter>()).setFromAnyValue(p);
             // }
 
-            MethodVisitor methodVisitor = visitor.visitMethod(Opcodes.ACC_PUBLIC, "set" + StringUtils.capitalize(property.getName()), RETURN_VOID_FROM_OBJECT, null, EMPTY_STRINGS);
+            MethodVisitor methodVisitor = visitor.visitMethod(Opcodes.ACC_PUBLIC, "set" + StringUtils.capitalize(getter.getName().substring(3)), RETURN_VOID_FROM_OBJECT, null, EMPTY_STRINGS);
```

Comment:
Can a getter also be called `isFoo()` instead of `getFoo()`?

Reply:
Excellent point @marcphilipp, according to the class inspection code, the getter method need to [explicitly return a `Boolean` type to match as an `IS_GETTER`](https://github.com/gradle/gradle/blob/master/subprojects/base-services/src/main/java/org/gradle/internal/reflect/PropertyAccessorType.java#L84-L87). It's worth adding a test to make this assumption stronger.